File: solutions/c/factorial.c
#include <stdio.h>

int fact(int n) {
    if (n==0) {
        return 1;
    } else {
        return n * fact(n-1);
    }
}

int main() {
    printf("%d\n", fact(0));
    printf("%d\n", fact(1));
    printf("%d\n", fact(5));
    return 0;
}

File: solutions/java/Factorial.java
class Factorial {
    public static int fact(int n) {
        if (n == 0) {
            return 1;
        } else {
            return n * fact(n-1);
        }
    }
    public static void main(String[] args) {
        System.out.println(fact(0));
        System.out.println(fact(1));
        System.out.println(fact(5));
    }
}